internal/hcl/modules/registry.go

```diff
@@ -137,6 +140,9 @@ func (d *Disco) DownloadLocation(moduleURL RegistryURL, version string) (string,
 	if err != nil {
 		return "", fmt.Errorf("unable to discover registry service using host %s %w", moduleURL.Host, err)
 	}
+	if !strings.HasSuffix(serviceURL.Path, "/") {
+		serviceURL.Path += "/"
+	}
```

Comment:
I thought about this second one a lot, not totally sure it's necessary, I think the resolver below was actually already account for mashing two paths together correctly.  But included this anyway for consistency